cilium/cmd/bpf_ct_list.go

```diff
@@ -69,9 +76,15 @@ func dumpCt(eID string) {
 			Fatalf("Unable to open %s: %s", path, err)
 		}
 		defer m.Close()
+		// Plain output prints immediately, JSON output holds until it
+		// collected values from all maps to have one consistent object
 		if command.OutputJSON() {
-			if err := command.PrintOutput(m); err != nil {
-				os.Exit(1)
+			callback := func(key bpf.MapKey, value bpf.MapValue) {
+				keyval := ctKeyVal{MapKey: key.(ctmap.CtKey).ToHost(), MapValue: *value.(*ctmap.CtEntry)}
```

Comment:
Why is the `.ToHost()` required? Was it broken before?

Reply:
It is not _required_ strictly speaking, but without it the `SourcePort` and `DestPort` from the key will be printed as they are stored in the BPF map, i.e. network byte order. It's not “broken” but not really convenient either, because you have to post-process the output to get the correct values for the ports.

Reply:
Hm, okay. Looks like there's the same issue with the `RevNat` field only in `CtEntry`.

Reply:
Ah, I missed it. There is no `ToHost()` available for the `CtEntry`, I'll see if I can add something.

Also looking into unit tests, but it requires mock maps or something.

Reply:
I'd be interested to hear what kind of mock map infrastructure you'd like. I've thought given the various maps we have, it'd be nice to have a generic infrastructure across all maps for testing purposes, eventually based upon github.com/cilium/ebpf . But I haven't looked at the problem recently enough to share any particularly useful insights.

Reply:
oh, and I see that the `test/runtime/cli.go` does have some runtime tests for the CLI, there may be an option there.